contrib/zenith/pagestore_smgr.c

```diff
@@ -106,6 +108,49 @@ typedef enum
 static SMgrRelation unlogged_build_rel = NULL;
 static UnloggedBuildPhase unlogged_build_phase = UNLOGGED_BUILD_NOT_IN_PROGRESS;
 
+
+/*
+ * Prefetch implementation:
+ * Prefetch is performed locally by each backend.
+ * There can be up to MAX_PREFETCH_REQUESTS registered using smgr_prefetch
+ * before smgr_read. All this requests are appended to primary smgr_read request.
+ * It is assumed that pages will be requested in prefetch order.
```

Comment:
Does this mean that you expect each prefetch request to be requested later, or that when you request a prefetched page, the prefetches before that request will not be requested and can be dropped?